data-raw/admiral_verify_templates.R

```diff
@@ -0,0 +1,238 @@
+# This script:  data-raw/admiral_verify_templates.R
+
+#
+# Generates ADaM from templates and compares to previously generated ADaM file (found in pharmaverseadam).
+# pharmaverseadam is the SOURCE.
+#
+# Much code taken from pharamavreseadam::create_adams_data.R
+# (https://github.com/pharmaverse/pharmaverseadam/blob/main/data-raw/create_adams_data.R)
+#
+# USAGE:   verify_templates()
+#
+# TODO:
+# - add `attr` to generated ADaM
+# - re-write `load_rda()`:  now serves two purposes and was written in simpler times
+# - where code overlaps, make pharamversadam script and this one the same.
+
+# Assumptions/Questions:
+# - ignore *.rda files in admiral/data (per Ben)
+# - written as standalone R script, not as package R function
+# - compares full ADaM - all rows (ie no reduction in number of rows in each dataset)
+# - use cli:: for messages/errors?
+
+#' (if were to add to `admiral` package)
+#' @param pkg  package (ex:  "admiral )
+#' @param adams_names ADaM or CDISC name, without prefix or suffix  (ex:  adlb)
+#' @param adams_new  character vector of ADaM after template is run
+#' @param adams_old  character vector of original ADaM done at earlier date, saved in github
+
+#' @param template_dir path  Directory where templates where package is installed (not sourced)
+#' @param cache_dir (cache)     ~/.config/R/admiral_templates_data   (varies by OS/configuration)
+
+#' Directories
+#' - admiral/data     not used
+#' - admiral/inst/templates/  templates to create ADaM datasets
+#' - cache_dir  cache, where newly generated ADaM files kept and diffdf reports
+#'      (os dependent)
+#' - adam_old_dir  - dir where prior ADaMs are downloaded and stored
+
+load_rda <- function(fileName) {
+  load(fileName)
+  get(ls()[ls() != "fileName"]) # returns dataset
+}
+save_rda <- function(data, file_path, new_name) {
+  # new_name must include  .rda
+  if (missing(new_name)) {
+    save(data, file = file_path, compress = "bzip2")
+  }
+}
+compare <- function(base, compare, keys, file=NULL) {
+  tryCatch(
+    {
+      diffdf::diffdf(
+        base = base,
+        compare = compare,
+        keys = keys,
+        file = file
+      )
+    },
+    error = function(e) {
+      message("Error in diffdf: ", e$message)
+    }
+  ) ## end tryCatch
+} ## end compare
+
+# start fresh
+clean_cache <- function() {
+  cache_dir <- tools::R_user_dir("admiral_templates_data", which = "cache")
+  if (dir.exists(cache_dir)) {
+    unlink(cache_dir, recursive = TRUE)
+    message("Cache directory deleted: ", cache_dir)
+  } else {
+    message("Cache directory does not exist: ", cache_dir)
+  }
+}
+
+clean_adam_old_dir = function(){
+  if (dir.exists(tempdir())) {
+    unlink(adam_old_dir, recursive = TRUE)
+    message(adam_old_dir,  "directory deleted: ", tempdir())
+  } else {
+    message(adam_old_dir, "directory does not exist: ", tempdir())
+  }
+}
+
+download_adam_old = function (adam_names){
```

Comment:
method is ok?

Reply:
do we need to clean the cache if this whole program just gets rerun again on each compare action?